app/scripts/modules/core/pipeline/config/pipelineConfigProvider.js

```diff
@@ -59,10 +59,45 @@ module.exports = angular.module('spinnaker.core.pipeline.config.configProvider',
       return angular.copy(stageTypes);
     }
 
-    function getConfigurableStageTypes() {
-      return getStageTypes().filter(function(stageType) {
-        return !stageType.synthetic && !stageType.provides;
+    function getConfigurableStageTypes(providers) {
+      let allStageTypes = getStageTypes();
```

Comment:
Is getStageTypes guaranteed to always return an array? If not we might want to default returning/assigning an empty array to prevent the downstream filter from bombing out.